[Muhammad Radhya Fawza] iP

mradhyaf opened the pull request:
# Kenobi
> "Hello there." - Obi-wan Kenobi - ([source](https://www.youtube.com/watch?v=WYGUKCINF_w))

Duke, but perhaps the archives are incomplete :confused: .

## Quick Start 
**1. Ensure you have java 11 or above**
In the terminal, type (without the '$'):
``` 
$ java --version
 ```
If it doesn't return the java version, [install java](https://www.wikihow.com/Install-Java)

**2. Download Kenobi binary** [here](https://github.com/mradhyaf/ip/releases) 
Optionally, build from [source](https://github.com/mradhyaf/ip)

**3. Open a terminal the same folder as the downloaded binary**
run `$ java --jar Kenobi`

## Basic commands
- `todo`, `deadline`, `event` add a todo, deadline, and event respectively
- `list` lists all tasks
- `delete n` deletes task at index _n_
- `mark n` / `unmark n` marks/unmarks a task as done at index _n_

Features:
- [x] Create, read, ~~update~~, delete tasks
- [ ] update tasks (coming soon)
- [ ] Obi-wan and grievous dialog (stretch goal)

#
Some Java code :tea:  :
```java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!")
    }
}
```

Commit:
Add Gradle support

mradhyaf committed:
Implement greeting, echo, and exit functionality

mradhyaf committed:
Implement add and list commands

mradhyaf committed:
Add ability to mark as done and implement Task class

mradhyaf committed:
Replace add command with todo, deadline, and event commands

mradhyaf committed:
Add minimal error handling and delete functionality

mradhyaf committed:
Run text-ui-test

mradhyaf committed:
Add TaskList class

mradhyaf committed:
Merge branch 'branch-level-7'

mradhyaf committed:
Add save functionality

mradhyaf committed:
Merge branch 'branch-level-7'

mradhyaf committed:
Abstracted storage, parse, ui, and command functionalities

mradhyaf committed:
Move classes into packages

mradhyaf committed:
Add date functionality, Refactor duke to kenobi, Move Parser into its own package and add ParseException

mradhyaf committed:
Merge branch 'branch-Level-8'

mradhyaf committed:
Add find functionality

mradhyaf committed:
Merge branch 'branch-Level-9'

mradhyaf committed:
Merge branch 'master' into A-CodingStandard

mradhyaf committed:
Refactor some code to conform with coding standards and Add some exception handling

mradhyaf committed:
Merge tag 'A-CodingStandard'

mradhyaf committed:
Merge branch 'master' into branch-A-JavaDoc

mradhyaf committed:
Move the find, mark, and unmark functionalities to TaskList and Add JavaDocs for everything except kenobi.command package

mradhyaf committed:
Add JavaDocs for classes in kenobi.command package

mradhyaf committed:
Merge branch 'branch-A-JavaDoc'

mradhyaf committed:
Improve exception handling in Parser class and Add some JUnit tests

mradhyaf committed:
Merge branch 'branch-A-JUnit'

mradhyaf committed:
Add untracked files that was supposed to be in the previous commit

mradhyaf committed:
Merge branch 'branch-A-JUnit'

mradhyaf committed:
Remove some unwanted files

mradhyaf committed:
Fix exception message

yl-ang commented on src/test/java/kenobi/parser/ParserTest.java:
```diff
@@ -0,0 +1,28 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ParserTest {
```
perhaps you can add header comments for the Junit test classes and methods.

yl-ang reviewed (commented):
Overall good job, the majority of the code follows the java coding standard.

yl-ang commented on src/main/java/kenobi/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+     * @return a Command object encapsulating the parsed command.
+     * @throws ParseException if the Parser could not parse the command
+     */
+    public static Command parse(String str) throws ParseException {
```
I like how you handle user input via the use of switch cases.

mradhyaf committed:
Remove typo

HanJiyao commented on src/main/java/kenobi/Kenobi.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    public static void main(String[] args) {
+        new Kenobi("../../../data/tasks.txt").run();
```
the path quite seems quite deep shoud it better with fewer parent folder?

HanJiyao commented on src/main/java/kenobi/command/Command.java:
```diff
@@ -0,0 +1,26 @@
+     *
+     * @param taskList The TaskList to which the command is to be executed.
+     */
+    public void setData(TaskList taskList) {
```
should it be setTaskList?

HanJiyao commented on src/main/java/kenobi/command/DeleteCommand.java:
```diff
@@ -0,0 +1,33 @@
+     * @param index
+     */
+    public DeleteCommand(int index) {
+        toDeleteIndex = index -1;
```
missing space for -

HanJiyao commented on src/main/java/kenobi/util/TaskList.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    public TaskList search(String searchTerm) {
+        TaskList tasks = (TaskList) clone();
+        tasks.removeIf(e -> !e.hasWord(searchTerm));
```
I like how you use lambda expressions to make funtion short and clear

HanJiyao commented on src/main/java/kenobi/util/TaskList.java:
```diff
@@ -0,0 +1,60 @@
+/**
+ * The TaskList encapsulates a list of Task objects.
+ */
+public class TaskList extends ArrayList<Task> {
```
I like how you make use of extend ArrayList

HanJiyao reviewed (commented):
Overall, I found your code mostly easy to follow. However, I did spot some minor coding standard violations. In addition maybe can have more JavaDoc comments to make more clear.

mradhyaf committed:
Merge branch 'add-gradle-support' of github.com:mradhyaf/ip

mradhyaf committed:
Set up Gradle minimally to build and run Kenobi as well as checkstyle

mradhyaf committed:
Add graphical user interface

mradhyaf committed:
Remove text ui test files

User interface for Kenobi has changed from text ui to graphical ui.

These test files were optimal for text ui but not gui.

As an effort to declutter the codebase, let's remove these text ui
files.

mradhyaf committed:
Fix missing javafx runtime components on fat jar build

Previously javafx.Application.launch method was called inside
a class extending javafx.Application class which seems to be
the cause of the bug.

To go around this bug, Launcher class is added and
javafx.Application.launch is called from here.

mradhyaf committed:
Add non-negative assertions for some commands

mradhyaf committed:
Merge pull request #2 from mradhyaf/branch-A-Assertions

Add some assertions

mradhyaf committed:
Replace invalid command handling with exception

mradhyaf committed:
Move load file parsing to Parser class

Parsing logic from save files are currently in Storage class.

To improve CodeQuality according to SLAP, let's move parsing logic to
Parser class.

mradhyaf committed:
Merge pull request #3 from mradhyaf/branch-A-CodeQuality

Improve code quality

mradhyaf committed:
Improve GUI

mradhyaf committed:
Add natural date extension

mradhyaf committed:
Add exit handling on GUI

mradhyaf committed:
Write user guide docs

mradhyaf committed:
Set theme jekyll-theme-cayman

mradhyaf committed:
Add UI screenshot

mradhyaf committed:
Merge branch 'master' of github.com:mradhyaf/ip

mradhyaf committed:
Fix dependency issues
